CentralUnit/Communication/XbeeNetwork.cs

```diff
@@ -0,0 +1,416 @@
+            }
+            else
+            {
+                this.highestFrameID++;
+                device = new XbeeSerialCommunication(address, this, this.highestFrameID);
+                this.devices.Add(device);
+            }
+
+            return device;
+        }
+
+        /// <summary>
+        /// Write data, to be used by <see cref="XbeeSerialCommunication"/> devices.
+        /// </summary>
+        /// <param name="data">The <c>Xbee</c> API data to write to the serial communication channel.</param>
+        internal void Write(byte[] data)
+        {
+            this.transmitQueue.Enqueue(data);
+        }
+
+        /// <summary>
+        /// Removes the <see cref="XbeeSerialCommunication"/> device from the network.
+        /// </summary>
+        /// <param name="address">The 64-bit address to identify the device to be removed.</param>
+        internal void CloseXbeeDevice(XBee64BitAddress address)
+        {
+            if (this.devices.Any(dev => dev.Xbee64address.Equals(address)))
+            {
+                this.devices.Remove(this.devices.First(dev => dev.Xbee64address.Equals(address)));
+            }
+        }
+
+        /// <summary>
+        /// Performs initialization tasks.
+        /// </summary>
+        /// <param name="communication">The communication channel to work with.</param>
+        private void Initialize(ISerialCommunication communication)
+        {
+            this.communicationChannel = communication;
+            this.communicationChannel.ConnectionStateChanged += this.CommunicationChannel_ConnectionStateChanged;
+            this.communicationChannel.DataReceived += this.CommunicationChannel_DataReceived;
+            this.communicationChannel.Failure += this.CommunicationChannel_Failure;
+            this.transmitQueue = new ConcurrentQueue<byte[]>();
+            this.receiveQueue = new ConcurrentQueue<byte[]>();
+            this.commThread = new Thread(this.RunXbeeNetwork) { IsBackground = true };
+            this.commThread.Start();
+            this.parser = new XBeePacketParser();
+        }
+
+        /// <summary>
+        /// Notifies all devices in this network of a failure in the serial communication.
+        /// </summary>
+        /// <param name="sender">The sender of the event</param>
+        /// <param name="e">EventArgs supplied with the event</param>
+        private void CommunicationChannel_Failure(object sender, EventArgs e)
+        {
+            foreach (XbeeSerialCommunication xbee in this.devices)
+            {
+                xbee.OnFailure();
+            }
+        }
+
+        /// <summary>
+        /// Processes data received from the communication channel.
+        /// </summary>
+        /// <param name="sender">The sender of the event</param>
+        /// <param name="e">EventArgs supplied with the event contains a byte array with data</param>
+        private void CommunicationChannel_DataReceived(object sender, DataReceivedEventArgs e)
```

Comment:
So far we've not used underscores in methods. What should our coding practice here be?

Im used to CamelCasing mehods. Unit tests are the exception since writing a descriptive name tend to get quite long

Reply:
This was generated by Visual Studio. Good question.